app/src/main/java/nextstep/shoppingcart/ui/AppNavHost.kt

```diff
@@ -18,10 +22,32 @@ internal fun AppNavHost(
         startDestination = Navigation.ProductList.route,
     ) {
         composable(Navigation.ProductList.route) {
-            ProductListScreen(navHostController)
+            ProductListScreen(
+                onCartClick = {
+                    navHostController.navigate(Navigation.Cart.route)
+                },
+                onProductItemClick = { productId ->
+                    navHostController.navigate(
+                        route = "${Navigation.ProductDetail.route}/${productId.id}"
+                    )
+                },
+            )
```

Comment:
반복될 수 있는 navigate 함수들은 별도의 확장 함수로 만들어 가독성을 높여보면 어떨까요? 일반적으로 다음과 같이 작성된 코드가 더 읽기 쉬울 것 같아요.
```suggestion
            ProductListScreen(
                onCartClick = {
                    navHostController.navigateToCart()
                },
                onProductItemClick = { product ->
                    navHostController.navigateToCart(product.id)
                },
            )
```

Reply:
nitpicking: Route를 선언할 때 굳이 중복을 제거하고 싶다면 다음과 같이 구현할 수 있습니다.
```kotlin
const val ROUTE = "${Navigation.ProductDetail.route}/{$PRODUCT_ID_ARG}"

// usage
ROUTE.replace("{$PRODUCT_ID_ARG}", product.id)
```
오히려 Route 구조를 파악하는 데에 포인트가 분산되어 가독성이 안좋아질 수도 있어서 개발자가 판단해서 구현하면 될 것 같습니다.